### DISCUSSION TITLE
SMTP is broken.

### DISCUSSION THREAD

**Discussion by CarlSinclair** [dani-garcia/vaultwarden#4833] [Q&A] [answered]
### Subject of the issue

Mail doesn't work. Error 111

### Deployment environment

* vaultwarden version: 1.31.0

* Install method: Explained [here](https://vaultwarden.discourse.group/t/404-on-initial-account-creation-without-docker/3838/4?u=test123)

* Clients used: Web vault, email 2FA
* Reverse proxy and version: Caddy v2.8.4 h1:q3pe0wpBj1OcHFZ3n/1nl4V4bxBrYoSoab7rL9BMYNk=
* MySQL/MariaDB or PostgreSQL version: Sqlite
* Other relevant details:

### Steps to reproduce

Nothing out of the ordinary. Email 2FA and password reset emails aren't working.
### Expected behaviour

Mail send
### Actual behaviour

Mail no send
### Troubleshooting data

[Here](https://vaultwarden.discourse.group/t/404-on-initial-account-creation-without-docker/3838/5?u=test123)

**Comment by BlackDex** [comment]
Looks like the container or your host just can't connect to the smtp server you provided. Trying all those methods will not help.

Either a firewall is blocking this or the container even can't connect to the internet.

Provide the Support String which you can generate via the Vaultwarden Admin Interface.

**Reply by BlackDex** [reply]
Are you sure the port is correct? It's set to 25. Must of the time it should be 587.

I would suggest to double-check the settings via the admin interface. If that doesn't work try to enable `SMTP_DEBUG=true` and see if that provides more details.

**Reply by CarlSinclair** [reply]
It is set to true. The only logs I've gotten are the ones I posted in that thread, and they're consistent no matter what port I try. I am successfully using these SMTP credentials on port 80 with another service. I've tried 587, 25, and 80 with Vaultwarden and they all lead to the exact same error.

**Reply by BlackDex** [reply]
Then check other logs on your system. Maybe some security tooling like SELinux or Apparmor is blocking stuff.

Vaultwarden is able to go to the internet and do DNS calls, so that part is working fine. The logs say `Connection Refused` so something is blocking, and that is outside of Vaultwarden as far as i can tell.

**Reply by dfunkt** [reply]
I'd also like to chime in to say that Vaultwarden works with Mailjet, that's what I was using until a few months ago. At that time I was using the alpine docker image with either caddy or cloudflare tunnels. (switching between them to check out how it works)

**Reply by BlackDex** [reply]
Also, you can try to use `tcpdump` or `wireshark` to test what Vaultwarden tries and see what happens.

**Reply by CarlSinclair** [reply]
> I'd also like to chime in to say that Vaultwarden works with Mailjet

I literally just switched FROM Mailjet because I couldn't get it to work with Vaultwarden and I had tried everything else.

> Maybe some security tooling like SELinux or Apparmor is blocking stuff.

I don't use either. That's why I'm convinced it's Vaultwarden because it can't be anything else.

I ran `tcpdump | grep mail` and got the following:

Which is interesting because I no longer have mailjet in my Vaultwarden `.env` file. Here's my SMTP config from `data/.env`:

The `SMTP_HOST` isn't even the only discrepancy. The config string also has other different values, which leads me to believe there's another config file somewhere being prioritized. Where should I look for this? The only directory path in my systemd service file is the data directory, and the environmental variable for the config file is this one. I have definitely restarted the service multiple times since these changes were made.

**Reply by BlackDex** [reply, marked as answer]
You have configured done settings via the admin interface. 
Which will override all other config options. 

This is shown during startup in the logs, and also in the support string you outputted.

So either delete the `config.json` from your data direct and make sure all settings are correct in your env file. Or update all settings via the admin interface and save it so those items are stored in the `config.json`.

**Reply by CarlSinclair** [reply]
It seems the problem was caused by the config files. All the changes I made in all of my tests (by modifying `.env` and restarting the service) weren't actually applying. All I did now was change the port to 80 from the admin page and it worked instantly. Very weird.

Thank you for your help. Maybe this should be mentioned somewhere, that there are multiple config files at play and that the correct place to change settings is via the admin interface.

**Reply by BlackDex** [reply]
Glad it works.
